apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/trace/ExitTypeSpan.java

```diff
@@ -18,8 +18,15 @@
 
 package org.apache.skywalking.apm.agent.core.context.trace;
 
-public interface WithPeerInfo {
+import org.apache.skywalking.apm.agent.core.context.ContextCarrier;
+
+/**
+ * The exit span has some additional behaviours
+ */
+public interface ExitTypeSpan {
     int getPeerId();
 
     String getPeer();
+
+    void inject(ContextCarrier carrier);
```

Comment:
Consider returning `this` for chaining

Reply:
After the tests pass, will do.

Reply:
From the API lever, I could only return `ExitTypeSpan`, is that useful?

Reply:
> From the API level, I could only return `ExitTypeSpan`, is that useful?

Yes, from the interface you have to return `ExitTypeSpan`, but you can return `ExitSpan` in its implementations, and chaining with the more specific types

<img width="752" alt="image" src="https://user-images.githubusercontent.com/15965696/76158658-263e0500-6153-11ea-84b5-c2153faffbaa.png">

Reply:
Done.